pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/ByteArrayBufferedData.java

```diff
@@ -84,10 +82,10 @@ public byte getByte(final long offset) {
     public long getBytes(final long offset, @NonNull final byte[] dst, final int dstOffset, final int maxLength) {
         validateLen(maxLength);
         final long len = Math.min(maxLength, length() - offset);
-        validateCanRead(offset, len);
         if (len == 0) {
             return 0;
         }
+        checkOffsetToRead(offset, length(), len);
```

Comment:
Since the offset is now checked after length, calls like `getBytes(-1, dst, 0, 0)` will now return 0, but threw an exception previously. I wonder if there are any tests for this behavior.

Reply:
I've moved this above the `if`, so now it will throw.